sunpy/time/time.py

```diff
@@ -215,54 +222,57 @@ def convert_time_str(time_string, **kwargs):
                 break
             if ts is None:
                 continue
-            return datetime.strptime(ts, time_format) + time_delta
+            return Time.strptime(ts, time_format, **kwargs) + time_delta
         except ValueError:
             pass
-    time_string_parse_format = kwargs.pop('_time_string_parse_format', None)
-    if time_string_parse_format is not None:
-        ts, time_delta = _regex_parse_time(time_string,
-                                           time_string_parse_format)
-        if ts and time_delta:
-            return datetime.strptime(ts, time_string_parse_format) + time_delta
-        else:
-            return datetime.strptime(time_string, time_string_parse_format)
+
     # when no format matches, call default fucntion
-    convert_time.dispatch(object)(time_string, **kwargs)
+    return convert_time.dispatch(object)(time_string, **kwargs)
 
 
-def parse_time(time_string, time_format='', **kwargs):
-    """Given a time string will parse and return a datetime object.
+def parse_time(time_string, *, format=None, **kwargs):
+    """Given a time string will parse and return a `astropy.time.Time` object.
     Similar to the anytim function in IDL.
     utime -- Time since epoch 1 Jan 1979
+
     Parameters
     ----------
-    time_string : [ int, float, time_string, datetime ]
-        Date to parse which can be either time_string, int, datetime object.
-    time_format : [ basestring, utime, datetime ]
+    time_string : [ int, float, string, datetime, astropy.time.Time,
```

Comment:
These should not be in `[]` and they should where possible be linked to their types.